### PR TITLE
feat: add popUp for password

### LINKED ISSUE #1540: Option for the user to type in password at device connection
**Actual Behaviour**

Currently the device credentials have hardcoded credentials for the device connection.

**Expected Behaviour**

Make a pop-up or dialog that allows the user to type the password and then the user's credentials are sent to the device.

**Steps to reproduce it**

Log in, click on the susi button , go to settings , scroll down to click to setup device and then scan for device and connect to the device.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by batbrain7**
Fixes #1540 

Changes: Added popup so that the user can enter the password for the successful device connection.

Screenshots for the change: 

e6a59453-3c72-4338-8a21-367d12e85250

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/data/DeviceModel.kt [hunk 1]
```diff
@@ -35,11 +38,13 @@ class DeviceModel : IDeviceModel {
...
             override fun onFailure(call: Call<String>?, t: Throwable?) {
                 if (t?.localizedMessage != null) {
-                    Timber.d(t.localizedMessage)
+                    Timber.d(t.cause)
                 } else {
                     Timber.d(t, "An error occurred")
                 }
-                listener.onSetConfigFailure()
+                Timber.d("Error in Configuration : " + call.toString())
+                if (t!=null)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/data/DeviceModel.kt]: Style error

File: app/src/main/java/org/fossasia/susi/ai/data/DeviceModel.kt [hunk 2]
```diff
@@ -58,7 +63,9 @@ class DeviceModel : IDeviceModel {
                 } else {
                     Timber.d(t, "An error occurred")
                 }
-                listener.onSendAuthFailure()
+                Timber.d("Error in Authentication : " + call.toString())
+                if (t!=null)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/data/DeviceModel.kt]: Style

File: app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt [hunk 3]
```diff
@@ -240,4 +239,25 @@ class DeviceConnectFragment : Fragment(), IDeviceConnectView {
         wifiList.adapter = recyclerAdapter
     }
 
+    override fun showPopUpDialog() {
+        val utilModel: UtilModel
+        utilModel = UtilModel(activity as DeviceActivity)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt]: Unnecessary assignment detached from declaration

**Comment by iamareebjamal** [comment]
@imjog Please see that code follows guideline before appoving

**Comment by iamareebjamal** [comment]
@batbrain7 Till when are we going to see these basic mistakes?

I'll have to automatically fail builds on code style violation which will make it very difficult for contributors

**Comment by imjog** [comment]
@iamareebjamal I approved PR just by seeing screenshot attached as I am not much familiar with Java/kotlin

**Comment by batbrain7** [comment]
@iamareebjamal I am sorry and I will be careful from now on that the code follows proper style.
I will send a PR that fixes style for the whole app.

**Pull request merged**